src/pages/[platform]/build-a-backend/data/working-with-files/index.mdx

```diff
@@ -700,7 +1060,1206 @@ One example is when we [create an API record, associate the Storage file with th
+        }
+    }
+
+    @ViewBuilder
+    func AdditionalOperations() -> some View {
+        if viewModel.currentSong != nil {
+            VStack {
+                Button("Refresh") {
+                    Task {
+                        try? await viewModel.refreshSongAndArtCover()
+                    }
+                }.buttonStyle(TappedButtonStyle())
+                Button("Remove association from song") {
+                    Task {
+                        try? await viewModel.removeImageAssociationFromSong()
+                    }
+                }.buttonStyle(TappedButtonStyle())
+                Button("Remove association and delete image") {
+                    Task {
+                        try? await viewModel.removeImageAssociationAndDeleteImage()
+                    }
+                }.buttonStyle(TappedButtonStyle())
+                Button("Delete song and art cover") {
+                    Task {
+                        try? await viewModel.deleteSongAndArtCover()
+                    }
+                    songName = ""
+                }.buttonStyle(TappedButtonStyle())
+            }.disabled(viewModel.isLoading)
+        }
+    }
+
+    @ViewBuilder
+    func IsLoadingView() -> some View {
+        if viewModel.isLoading {
+            ZStack {
+                DimmedBackgroundView()
+                ProgressView()
+            }
+        }
+    }
+}
+
+struct SongView_Previews: PreviewProvider {
+    static var previews: some View {
+        SongView()
+    }
+}
+```
+</Block>
+<Block name="Photo Album">
+```swift title="PhotoAlbumView"
+import SwiftUI
+import Amplify
+import Photos
+
+class PhotoAlbumViewModel: ObservableObject {
+    @Published var currentImages: [UIImage] = []
+    @Published var currentAlbum: PhotoAlbum? = nil
+    @Published var isLoading: Bool = false
+
+    // Create a record with multiple associated files
+    func createPhotoAlbum(name: String, photos: [UIImage]) async throws {
+        await setIsLoading(true)
+        defer {
+            Task {
+                await setIsLoading(false)
+            }
+        }
+
+        let imagesData = photos.compactMap { $0.pngData() }
+        guard !imagesData.isEmpty else {
+            print("Could not get data from [UIImage]")
+            return
+        }
+
+        // Create the photo album record
+        let album = PhotoAlbum(name: name)
+        let result = try await Amplify.API.mutate(request: .create(album))
+        guard case .success(var createdAlbum) = result else {
+            print("Failed with error: ", result)
+            return
+        }
+
+        // Upload the photo album images
+        let imageKeys = await withTaskGroup(of: String?.self) { group in
+            for imageData in imagesData {
+                group.addTask {
+                    let key = "\(album.id)-\(UUID().uuidString)"
+                    do {
+                        _ = try await Amplify.Storage.uploadData(key: key,
```

Comment:
the examples above are pulled from the full code snippet, this will also need to be updated to use path